Alien Nuclear Operatives (Again) + Welding Shield Eyes as Uplink items

SgtHunk opened the pull request:
## About The Pull Request

This allows non-human races to become Nuclear Operatives (instead of being forced into human)
And I also added Welding Shield Eyes in a one-use Autosurgeon Syndicate implant, for traitors in case they want a more stealthy approach regarding their flash protection. _Or don't want to get fucked over by flashes because they bought thermals._

## Why It's Good For The Game

![image](https://user-images.githubusercontent.com/68669754/91613752-25da3600-e94e-11ea-808c-d29e81e6494f.png)
![image](https://user-images.githubusercontent.com/68669754/91613760-296dbd00-e94e-11ea-9c0a-6c251c2dc5ed.png)
![image](https://user-images.githubusercontent.com/68669754/92336713-e9a78380-f079-11ea-83bf-0121d3aa2f1f.png)


Beefmen Ops will be real, and Traitors/Nukies will be able to become flashproof while still keeping their stealth certain to a degree.

## Changelog
:cl:
add: Added support for Alien Nuclear Operatives. (They spawn with Iron/Salbutamol because otherwise beefmen/plasmemes will die quicker. Don't blame me, blame Franklin.
add: Added Plasmeme hardsuits, more costly, limited stock but pressureproof and no slowdown. Now you won't burn to death on Combat mode, Operative.
imageadd: added Plasmeme Syndicate envirosuits, and Plasmeme Syndie Hardsuits.
/:cl:

SgtHunk committed:
I hate beefmen

The whole thing, alien ops + welding shield eyes implant

GuillaumePrata commented:
Plasmamen aren't spawning properly with the internals then?
And Beefman probably need a fix for how they work too, but I don't understand much about their mechanics.

SgtHunk commented:
Plasmemes spawn properly, the main issue is the leader (He has both hands busy because of the war declare + nuke code papers, but normal plasmeme operatives spawn with the plasma tank in hand activated so those work)
That's the main reason for the salbutamol, so they can activate internals

ATH1909 commented:
Playing a plasmaman nukie looks like it'd be absolutely hellish, what with your 1.5x brute and burn multipliers and all.

SgtHunk commented:
You live by the plasmeme life you die by the plasmeme life

ATH1909 commented:
Will a plasmaman who's wearing a combat hardsuit catch on fire if they switch from EVA mode to combat mode?

SgtHunk commented:
Does seem like it, considering how to toggle EVA/Combat you need a helmet up which does fuck up plasmemes more
Honestly any antag with plasmemes just ends up being hard mode

ATH1909 commented:
> Does seem like it, considering how to toggle EVA/Combat you need a helmet up which does fuck up plasmemes more
> Honestly any antag with plasmemes just ends up being hard mode

I mean, I think they'd just take their robotics helmet off and then flip their combat hardsuit's helmet on while under a shower.

What I'm asking is: Once you have your combat hardsuit's helmet up, do you have to keep it in EVA mode 24/7 to avoid catching on fire, or will combat mode still protect you?

ATH1909 commented:
And there's a difference between "hard mode" and "has a 1.5x brute and burn modifier, as well as either hardsuit slowdown (making them practically sitting ducks to literally anyone with a ranged weapon) or basically no armor (if they opt to forgo using their combat hardsuit entirely)".

SgtHunk commented:
oh god combat mode doesn't offer the protection to make plasmemes not catch on fire (and this getting set on fire bypasses the elite hardsuit's temp-proofing so plasmemes are fucked either way)
will see if I can code in plasmeme-exclusive hardsuits to compensate for this

SgtHunk committed:
Plasmeme support baybee

SOOOOOO, what this does is:
Adds two new uplink items: Plasmaman Syndicate Hardsuits.
They are more costly, however they provide innate space protection and no slowdown.
Also adds a paper that warns plasmemes that the normal hardsuits won't work.
Oh yeah and also unfucks my adding it on the .dme using / instead of \

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+	name = "Syndicate Leader Plasmaman - Basic"
+	head = /obj/item/clothing/head/helmet/space/plasmaman/robotics/syndicate
+	uniform = /obj/item/clothing/under/plasmaman/security/syndicate
+	r_pocket= /obj/item/tank/internals/plasmaman/belt/full // They spawn with both hands busy, so they have to manually set internals
```
make sure that this won't overwrite (or be overwritten by) anything important

SgtHunk replied:
the war declare + paper are spawned on hand, which is why i was forced to move internals to r_pocket
(And pretty sure it's empty, left pocket having the makarov and the other items being on backpack)

ATH1909 replied:
you might want to check that quirks or something can't replace that right pocket slot's item

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+
+/obj/item/clothing/under/plasmaman/security/syndicate
+	name = "syndicate plasma envirosuit"
+	desc = "A non-descript and slightly suspicious looking plasmaman enviromental suit."
```
```suggestion
	desc = "A nondescript and slightly suspicious looking plasmaman enviromental suit."
```

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+	worn_icon = 'icons/fulpicons/phoenix_nest/syndicate_suits.dmi'
+	icon = 'icons/fulpicons/phoenix_nest/syndicate_suits_icons.dmi'
+	icon_state = "syndicate_envirosuit"
+	has_sensor = NO_SENSORS
```
make sure that this doesn't inherit bodycams either (do plasmaman seccies even have those?)

ATH1909 replied:
hell, since you're overwriting the armor values and the sprites anyway, you might as well make this a subtype of normal plasmaman jumpsuits instead of sec ones.

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+
+/obj/item/clothing/head/helmet/space/hardsuit/syndiplasmeme
+	name = "plasmaman blood-red hardsuit helmet"
+	desc = "An advanced helmet designed for work in special operations. Property of the Plasmaman Division of Gorlex Marauders."
```
is it "the Gorlex Marauders" or just "Gorlex Marauders"? I dunno, and I'm too lazy to check for myself, but you might wanna check existing mentionings of them in the code, just to be sure.

SgtHunk replied:
its simply "Gorlex Marauders" taking from the same description the blood-red hardsuit has

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+	name = "WARNING - Plasmaman Nuclear Operative"
+	info = {"<center><b>Attention, Operative</b></center>Operative. We are sorry to inform that due to certain budget cuts regarding Gorlex Marauders,
+		our standard-issue hardsuits aren't compatible with our kind. To make up for this, the Plasmaman Division has engineered two special hardsuits
+		that are available exclusively to you, on your uplink.
```
```suggestion
		that are available exclusively to your kind, which are available to be purchased from your uplink.
```

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+	info = {"<center><b>Attention, Operative</b></center>Operative. We are sorry to inform that due to certain budget cuts regarding Gorlex Marauders,
+		our standard-issue hardsuits aren't compatible with our kind. To make up for this, the Plasmaman Division has engineered two special hardsuits
+		that are available exclusively to you, on your uplink.
+		<br>These hardsuits will always protect you from pressure, and removing the mechanisms that let you
```
```suggestion
		<br>These hardsuits will always protect you from pressure, and by removing the mechanisms that let you
```

ATH1909 commented on code/FulpstationCode/fulp_ops.dm:
```diff
@@ -0,0 +1,107 @@
+		that are available exclusively to you, on your uplink.
+		<br>These hardsuits will always protect you from pressure, and removing the mechanisms that let you
+		change between combat mode and space-travel mode, we were able to remove the slowdown that the normal hardsuits give.
+		<br>Using the standard-issued hardsuits	will most likely end with you in flames after you switch to Combat Mode.
```
```suggestion
		<br>Attempting to use the standard-issue hardsuits will most likely result in you bursting into flames after you switch to Combat Mode.
```

ATH1909 commented on code/modules/uplink/uplink_items_fulp.dm:
```diff
@@ -101,10 +101,38 @@
+	cost = 6 // The normal Space Suit costs 4TC, with the Elite hardsuit priced at the double of the Space suit.
+	item = /obj/item/clothing/suit/space/hardsuit/syndiplasmeme
+	restricted_species = list("plasmaman")
+	limited_stock = 1 // No, we're not letting you give your entire nuke team no-slowdown EVA hardsuits. They can buy Elite for 2 more TC, anyways.
```
a plasmaman operative could theoretically buy one of these hardsuits for themselves, then buy a plasmaman elite syndicate hardsuit for one of their non-plasmaman buddies

SgtHunk replied:
correct, which i'd just let pass imo.
if a plasma man is crippling themselves to give another teammate another advantage, then i'd let them

ATH1909 commented:
> oh god combat mode doesn't offer the protection to make plasmemes not catch on fire (and this getting set on fire bypasses the elite hardsuit's temp-proofing so plasmemes are fucked either way)
> will see if I can code in plasmeme-exclusive hardsuits to compensate for this

Nice, thanks.

Thought here- what if, instead of adding a new hardsuit for plasmaman ops, you just let syndicate masks count as a plasmaman helmet for plasmaman "do I burst into flames" checks?

Like, you could go to this file https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/human/species_types/plasmamen.dm and change these two lines
```
	var/atmos_sealed = CanIgniteMob(H) && (isclothing(H.wear_suit) && H.wear_suit.clothing_flags & STOPSPRESSUREDAMAGE) && (isclothing(H.head) && H.head.clothing_flags & STOPSPRESSUREDAMAGE)
	if(!atmos_sealed && (!istype(H.w_uniform, /obj/item/clothing/under/plasmaman) || !istype(H.head, /obj/item/clothing/head/helmet/space/plasmaman) || !istype(H.gloves, /obj/item/clothing/gloves)))
```
to be
```
	var/atmos_sealed = CanIgniteMob(H) && (isclothing(H.wear_suit) && H.wear_suit.clothing_flags & STOPSPRESSUREDAMAGE) && ((isclothing(H.head) && H.head.clothing_flags & STOPSPRESSUREDAMAGE) || istype(H.wear_mask, /obj/item/clothing/mask/gas/syndicate))
	if(!atmos_sealed && (!istype(H.w_uniform, /obj/item/clothing/under/plasmaman) || (!istype(H.head, /obj/item/clothing/head/helmet/space/plasmaman) && !istype(H.wear_mask, /obj/item/clothing/mask/gas/syndicate)) || !istype(H.gloves, /obj/item/clothing/gloves)))
```
.

SgtHunk commented:
that works, I'll try to implement it along with some slight changes towards the envirosuit + the fix
